src/context/filter.rs

```diff
@@ -0,0 +1,108 @@
+use crate::{
+    Context, RedisModuleCommandFilter, RedisModuleCommandFilterCtx, RedisModuleString,
+    RedisModule_CommandFilterArgDelete, RedisModule_CommandFilterArgGet,
+    RedisModule_CommandFilterArgInsert, RedisModule_CommandFilterArgReplace,
+    RedisModule_CommandFilterArgsCount, RedisModule_CommandFilterGetClientId,
+    RedisModule_RegisterCommandFilter, RedisModule_UnregisterCommandFilter, ValkeyString,
+};
+use std::ffi::c_int;
+use std::str::Utf8Error;
+
+#[derive(Debug, Clone, Copy)]
+pub struct CommandFilter {
+    inner: *mut RedisModuleCommandFilter,
+}
+
+// otherwise you get error:     cannot be shared between threads safely
+unsafe impl Send for CommandFilter {}
+unsafe impl Sync for CommandFilter {}
+
+impl CommandFilter {
+    pub fn new(inner: *mut RedisModuleCommandFilter) -> Self {
+        CommandFilter { inner }
+    }
+
+    pub fn is_null(&self) -> bool {
+        self.inner.is_null()
+    }
+}
+
+pub struct CommandFilterCtx {
+    inner: *mut RedisModuleCommandFilterCtx,
+}
+
+impl CommandFilterCtx {
+    pub fn new(inner: *mut RedisModuleCommandFilterCtx) -> Self {
+        CommandFilterCtx { inner }
+    }
+
+    pub fn args_count(&self) -> c_int {
+        unsafe { RedisModule_CommandFilterArgsCount.unwrap()(self.inner) }
+    }
+
+    pub fn arg_get(&self, pos: c_int) -> *mut RedisModuleString {
+        unsafe { RedisModule_CommandFilterArgGet.unwrap()(self.inner, pos) }
+    }
+
+    // helper to get argument as a &str instead of RedisModuleString
+    pub fn arg_get_as_str<'a>(&self, pos: c_int) -> Result<&'a str, Utf8Error> {
+        let arg = self.arg_get(pos);
+        ValkeyString::from_ptr(arg)
+    }
+
+    // helper to grab 0 argument, the command which is always present and return as str
+    pub fn cmd_get_as_str(&self) -> &str {
+        let cmd = self.arg_get(0);
+        ValkeyString::from_ptr(cmd).unwrap()
+    }
+
+    // helper to get all args minus the command (0th arg)
+    pub fn get_all_args_wo_cmd(&self) -> Vec<&str> {
+        let mut output = Vec::new();
+        for pos in 1..self.args_count() {
+            match self.arg_get_as_str(pos) {
+                Ok(arg) => output.push(arg),
+                Err(_) => continue, // skip invalid args
+            }
+        }
+        output
+    }
+
+    pub fn arg_replace(&self, pos: c_int, arg: *mut RedisModuleString) {
+        unsafe { RedisModule_CommandFilterArgReplace.unwrap()(self.inner, pos, arg) };
+    }
+
+    pub fn arg_insert(&self, pos: c_int, arg: *mut RedisModuleString) {
+        unsafe { RedisModule_CommandFilterArgInsert.unwrap()(self.inner, pos, arg) };
+    }
+
+    pub fn arg_delete(&self, pos: c_int) {
+        unsafe { RedisModule_CommandFilterArgDelete.unwrap()(self.inner, pos) };
+    }
+
+    pub fn get_client_id(&self) -> u64 {
+        unsafe { RedisModule_CommandFilterGetClientId.unwrap()(self.inner) }
+    }
+}
+
+impl Context {
```

Comment:
Should we add documentation style comments for the two functions below (so that it shows up in the Rust Documentation)?

Reply:
added